app/uk/gov/hmrc/helptosave/services/BarsService.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosave.services
+
+import java.util.UUID
+
+import cats.instances.string._
+import cats.syntax.eq._
+import com.google.inject.{ImplementedBy, Inject, Singleton}
+import play.api.http.Status
+import uk.gov.hmrc.helptosave.audit.HTSAuditor
+import uk.gov.hmrc.helptosave.config.AppConfig
+import uk.gov.hmrc.helptosave.connectors.BarsConnector
+import uk.gov.hmrc.helptosave.metrics.Metrics
+import uk.gov.hmrc.helptosave.models.{BARSCheck, BarsRequest}
+import uk.gov.hmrc.helptosave.util.Logging.LoggerOps
+import uk.gov.hmrc.helptosave.util.{LogMessageTransformer, Logging, NINO, PagerDutyAlerting}
+import uk.gov.hmrc.http.HeaderCarrier
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@ImplementedBy(classOf[BarsServiceImpl])
+trait BarsService {
+
+  type BarsResponseType = Future[Either[String, Boolean]]
+
+  def validate(barsRequest: BarsRequest, path: String)(implicit hc: HeaderCarrier, ec: ExecutionContext): BarsResponseType
```

Comment:
you should be able t get the path from the header carrier somewhere - you shouldn't need to pass it in as a separate parameter here

Reply:
we can get it from `Request` but not sure if its a good idea to deal with Request entities in the services layer , but i can do it

Reply:
if that's the case, there's no point in making that change as we'll just be swapping one input for another. It's just slightly annoying the info isn't there in the HeaderCarrier which we're passing in anyway